### DISCUSSION TITLE
502 Bad Gateway

### DISCUSSION THREAD

**Discussion by thremilien** [firefly-iii/firefly-iii#5914] [Firefly III]
Hello, I have a problem with fidi. On his ip it works very well, I tried to add a reverse proxy and now I have a 502 bad gateway...
Maybe the solution is very simple but I can't find... Don't hesitate if you need my .env or conf files or more info, I don't know what to give ^^'

![image](https://user-images.githubusercontent.com/51836065/159155668-c7884110-3706-4db5-a62b-3924e4cb50c5.png)

**Comment by JC5** [comment]
A 502 is a very generic error and usually means that your openresty config is pointing to the wrong host + port. What is your proxy_pass value?

**Reply by thremilien** [reply]
I use Nginx proxy manager this is my configuration. (Tell me if that wasn't what you wanted)
![image](https://user-images.githubusercontent.com/51836065/159157500-c75b2e0e-e349-43b1-9e7e-abbbf2580f7c.png)
![image](https://user-images.githubusercontent.com/51836065/159157503-ee3fdaf6-6c05-443d-b5ee-8af1c9d85583.png)
![image](https://user-images.githubusercontent.com/51836065/159157512-c3e77500-59d6-4943-97a5-4a171a8a8af6.png)
![image](https://user-images.githubusercontent.com/51836065/159157515-ea1a50fe-f5a1-4b13-b476-d7e416fdf8c1.png)

**Reply by JC5** [reply]
It looks OK, but Firefly III runs on port 8080 by default. Try that instead of 80.

**Reply by thremilien** [reply]
It's fidi (firefly data importer) and when I access it with his ip I use port 80... (I try 8080 but it isn't working anymore). I don't use docker, I've deployed myself firefly and fidi

**Reply by JC5** [reply]
They both run on port 8080, so that doesn't matter.

**Reply by thremilien** [reply]
I didn't deploy fidi (nor firefly) with docker, I did it with apache so the base port is just 80 (the proof when I access it via the ip I don't specify ports). I still tested port 8080 just in case but it's also 502 Bad Gateway. 
![image](https://user-images.githubusercontent.com/51836065/159160642-7686203a-7607-4dd9-a0cf-7e73ae3cc212.png)
https://nginx.e-thor.fr/ is my reverse proxy address.

When I access it via its ip:
![image](https://user-images.githubusercontent.com/51836065/159160700-2a628c21-0cb6-47c7-a1a8-e3b3bc4d2fd2.png)
![image](https://user-images.githubusercontent.com/51836065/159160722-ec08e9f6-72ea-48ab-88d1-4f8e9e021ea1.png)

**Reply by JC5** [reply]
Sorry for the confusion. But in that case, I'm kind of lost.

Are you sure that your host (where the reverse proxy runs) can also reach 10.0.0.115? Have you tried ping or curl from the reverse proxy?

**Reply by thremilien** [reply]
Yes my reverse proxy can reach fidi 
![image](https://user-images.githubusercontent.com/51836065/159161304-96b30b1d-3f80-4583-881b-14e3d3cbc991.png)
Maybe it's an apache issues ?

**Reply by Jasper-Dev** [reply]
Hi,

I'm experiencing the same problem, only difference is that i am hosting everything on docker. 

![afbeelding](https://user-images.githubusercontent.com/33641245/171535442-45e07ac5-3ce3-458c-a928-9dbb6d9bfe62.png)
_Problem when trying to reach importer_

![afbeelding](https://user-images.githubusercontent.com/33641245/171535791-6d7fed4a-4776-48d3-8a5b-559b273c6e45.png)
_firefly instance does work_

![afbeelding](https://user-images.githubusercontent.com/33641245/171539659-fbac3e28-4e04-4239-9f3a-be8eefde7c3d.png)
all containers are running

![afbeelding](https://user-images.githubusercontent.com/33641245/171535969-27c8bf69-6d58-4be7-9de5-b24dfb4ba49e.png)
_everything is in the same docker network_

![afbeelding](https://user-images.githubusercontent.com/33641245/171537709-4e30b7c1-17f9-4b07-b278-bf5602dd2dec.png)
_reverse proxy runs_

![afbeelding](https://user-images.githubusercontent.com/33641245/171537436-c6e974c2-df69-43f4-b16e-9ec0f501c764.png)
_ping from nginx proxy manager to importer internal docker ip_

![afbeelding](https://user-images.githubusercontent.com/33641245/171540265-ee6fbf01-a039-4f67-a481-15535743c3d1.png)
_accessing the importer via external docker ip and external port_


Anyone got any idea?

**Reply by JC5** [reply]
You have the same issue and the same solution, because you're also using nginx.

**Reply by Jasper-Dev** [reply]
> You have the same issue and the same solution, because you're also using nginx.

I figured as much, hence why I reacted to this thread 😅

Plus, I was hoping OP has found a solution in the meantime and is willing to share, or that my situation is similar yet different enough to further narrow down the source of the problem

**Reply by JC5** [reply]
Sorry, I got confused with https://github.com/orgs/firefly-iii/discussions/6119 

That may be the fix!

**Reply by Jasper-Dev** [reply]
> Sorry, I got confused with https://github.com/orgs/firefly-iii/discussions/6119
> 
> That may be the fix!

Yeah I did come across that tread aswell, and maybe it can solve OP's problem, but unfortunately doesn't solve mine.

However, i did find something interesting.. 

![afbeelding](https://user-images.githubusercontent.com/33641245/171553903-8c0ea351-dc78-4e39-abe7-6231822d3868.png)
_when accessing the url and appending it with the external port, it seems to work_

![afbeelding](https://user-images.githubusercontent.com/33641245/171554057-340c25d1-d866-4991-8952-97e617dbf6c2.png)
_funny thing is, once it's loaded into the cache I can access it without specifying the port_

![afbeelding](https://user-images.githubusercontent.com/33641245/171554191-e637882a-0cf7-4728-b7f6-f00633ccdbac.png)
_untill i clear the cache and then its back to the problem_

**Reply by Jasper-Dev** [reply]
> > Sorry, I got confused with https://github.com/orgs/firefly-iii/discussions/6119
> > That may be the fix!
> 
> Yeah I did come across that tread aswell, and maybe it can solve OP's problem, but unfortunately doesn't solve mine.

Okay so, this seems to fix my problem after all! 🥳

Apparently, instead of creating a custom '/' location, I just needed to put the lines into the Advanced tab.. 😅

![afbeelding](https://user-images.githubusercontent.com/33641245/171641723-2555e167-676a-4174-aa9f-d26c3e938235.png)

```
client_max_body_size	64M;
proxy_redirect          off;
proxy_buffer_size       128k;
proxy_buffers           4 256k;
proxy_busy_buffers_size 256k;
```

and then it works! :D

![afbeelding](https://user-images.githubusercontent.com/33641245/171642708-6549acb3-aa52-4266-9076-0aa09f1e246d.png)



I hope this solution will solve @Emilien212 problem aswell!

@JC5 thanks for the help!

and @[sanderisbestok](https://github.com/sanderisbestok) for the initial solution in https://github.com/orgs/firefly-iii/discussions/6119#discussioncomment-2860630

**Comment by hanisntsolo** [comment]
I was also facing the same issue using my docker-compose. I changed to the above now im good.
version: '3.8'

services:
  firefly:
    image: fireflyiii/core:latest
    container_name: firefly
    restart: always
    ports:
      - "1082:8080"
    volumes:
      - ./docker/firefly/data:/var/www/firefly-iii/storage
      - ./docker/firefly/firefly_iii_upload:/var/www/html/storage/upload 
    environment:
      - APP_KEY=ZhTIN4aDoQo66uOPxqk8c8jMY9H6AVrE
      - DB_CONNECTION=mysql
      - DB_HOST=db
      - DB_DATABASE=firefly
      - DB_USERNAME=firefly
      - DB_PASSWORD=zaq12wsx
      - DOCKER_HUB_USERNAME=${DOCKER_HUB_USERNAME}
      - DOCKER_HUB_PASSWORD=${DOCKER_HUB_PASSWORD}
    depends_on:
      - db
    networks:
      - firefly_network

  db:
    image: mysql:5.7
    container_name: firefly_db
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: zaq12wsx
      MYSQL_DATABASE: firefly
      MYSQL_USER: firefly
      MYSQL_PASSWORD: zaq12wsx
    volumes:
      - ./docker/firefly/db_data:/var/lib/mysql
    ports:
      - "3308:3306"
    networks:
      - firefly_network

networks:
  firefly_network:
    driver: bridge
